container-engine-lib/lib/backend_impls/kubernetes/kubernetes_kurtosis_backend/engine_functions/create_engine.go

```diff
@@ -149,7 +154,29 @@ func CreateEngine(
 		}
 	}()
 
-	enginePod, enginePodLabels, err := createEnginePod(ctx, namespaceName, engineAttributesProvider, imageOrgAndRepo, imageVersionTag, envVars, privatePortSpecs, serviceAccount.Name, kubernetesManager)
+	// if engine node specified, label node with engine node name so engine node gets schedule on this node via node selectors passed to create pod
```

Comment:
add why engine node name is needed here (engine needs to get scheduled on same node for logs)